### ISSUE TITLE
AV1522: False positive on 'with' expressions (nondestructive mutation)

### ISSUE THREAD

**Issue by MaximeSmissen** [bkoelman/CSharpGuidelinesAnalyzer#132]
#### Analyzer

Assign each variable in a separate statement (AV1522)

#### Repro steps

1. Create a [record type](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record);
2. Create an object of the record type;
3. Using a 'with' expression (aka [nondestructive mutation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record#nondestructive-mutation)) make a new object of the record, modifying at least one property;

#### Expected behavior

No warning, because the whole expression is a single assignment to the newly created object.

#### Actual behavior

A warning is reported that the Class.Property and variable are both assigned in a single statement.

#### Example from the linked page

```csharp
public record Person(string FirstName, string LastName);

public static void Main()
{
    Person person1 = new("Nancy", "Davolio");
    Console.WriteLine(person1);
    // output: Person { FirstName = Nancy, LastName = Davolio }

    Person person2 = person1 with { FirstName = "John" }; // warning here
    Console.WriteLine(person2);
    // output: Person { FirstName = John, LastName = Davolio }
}
```
Warning at 'warning here' states: 'person2' and 'Person.FirstName' are assigned in a single statement

**Comment by bkoelman** [comment]
Hi @MaximeSmissen, thanks for taking the time to report this.
It is a bug indeed, I've created a PR to fix this.

**Comment by MaximevanderSmissen** [comment]
If I may be so bold as to ask @bkoelman, when can I expect a new release that includes this fix to be created?
My team and I are actively beginning to use these new records and related features, and I'd prefer not to have to ignore this issue on every occasion.
In any case, keep up the great work on this handy tool, it's greatly appreciated!

**Comment by bkoelman** [comment]
Hi @MaximeSmissen, apologies for not releasing this sooner. At the time, I wanted to make some other changes as well in the next few days, but got overwhelmed by other things and totally forgot about it.

[v3.8.2](https://github.com/bkoelman/CSharpGuidelinesAnalyzer/releases/tag/v3.8.2) has been released now. Sorry for the inconvenience.

**Comment by MaximeSmissen** [comment]
Thanks a lot for the quick response and release @bkoelman!
No need to apologize at all, everyone is busy at times and we're already very happy to be able to use this.
